datafusion/src/physical_plan/planner.rs

```diff
@@ -265,9 +266,48 @@ impl DefaultPhysicalPlanner {
                 }
 
                 let input_exec = self.create_initial_plan(input, ctx_state)?;
-                let physical_input_schema = input_exec.schema();
-                let logical_input_schema = input.as_ref().schema();
 
+                // at this moment we are guaranteed by the logical planner
+                // to have all the window_expr to have equal sort key
```

Comment:
Is the design that the planner will group all WindowExpr that have the same window (ORDER BY, PARTITION BY) into the same `LogicalPlan:Window` node? That sounds like a good plan to me 👍 

It might (eventually) be worth validating that invariant here and `assert`ing (or returning an Error` if the sort  keys are not the same for multiple window exprs (mostly to catch potential bugs faster in the future)

Reply:
i can add the assertion later. yes the invariant is real.